### PR TITLE
Backend/entries

### LINKED ISSUE #30: Implement Entry Service

### PR DESCRIPTION
**Pull request by okidokidokkaebi**
Please review and merge #51 first.

Resolves #30

### CODE REVIEW

File: src/main/java/com/peerrequest/app/services/EntryServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        if (optional.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var entry = optional.get();
+
+        if (newProps.name() != null) {
+            entry.setName(newProps.name());
+        }
+
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/services/EntryServiceImpl.java:46]: update of the categoryID is missing

File: src/main/java/com/peerrequest/app/api/EntriesController.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    @PatchMapping("/categories/{category_id}/entries")
+    Entry.Dto patchEntries(@RequestBody Entry.Dto dto, @AuthenticationPrincipal OAuth2User user) {
+        if (dto.id().isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "id is required");
+        }
+
+        if (dto.researcherId().isPresent()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "you may not change the researcher_id");
+        }
+
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/EntriesController.java:82]: check is missing if document id is present

File: src/main/java/com/peerrequest/app/api/EntriesController.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        if (!patchEntry.get().getResearcherId().equals(user.getAttribute("sub"))) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Only the user that created the entry may patch it");
+        }
+
+        var option = this.entryService.update(dto.id().get(), dto);
+
+        return option.get().toDto();
+    }
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/EntriesController.java:97]: Delete entry is missing

**Review by Wil-Lee — CHANGES_REQUESTED** [review]

**Comment by bits0rcerer** [comment]
Note: rebased `backend/entries` onto `backend/categories`

File: src/main/java/com/peerrequest/app/services/EntryServiceImpl.java (hunk 1 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/services/EntryServiceImpl.java:46]: Dropped FR.

File: src/main/java/com/peerrequest/app/api/EntriesController.java (hunk 2 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/EntriesController.java:82]: Added the check.

File: src/main/java/com/peerrequest/app/api/EntriesController.java (hunk 3 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/EntriesController.java:97]: Added the endpoint.

**Review by Wil-Lee — DISMISSED** [review]

**Review by bits0rcerer — DISMISSED** [review]

**Review by Wil-Lee — APPROVED** [review]

**Pull request merged**